tests/lib/nested.sh

```diff
@@ -205,18 +266,66 @@ start_nested_core_vm(){
+    cp -f "$WORK_DIR/image/ubuntu-core.img" "$IMAGE_FILE"
+
+    # Now qemu parameters are defined
+    PARAM_MEM="-m 2048"
+    PARAM_DISPLAY="-nographic"
+    PARAM_EXTRA="-machine accel=kvm"
+    PARAM_NETWORK="-net nic,model=virtio -net user,hostfwd=tcp::$SSH_PORT-:22"
+    PARAM_ASSERTIONS="-drive file=$WORK_DIR/assertions.disk,cache=none,format=raw"
+    PARAM_MONITOR="-monitor tcp:127.0.0.1:$MON_PORT,server,nowait -usb"
+    if is_core_20_nested_system; then
+        if ! is_focal_system; then
+            cp /etc/apt/sources.list /etc/apt/sources.list.back
+            echo "deb http://us-east1.gce.archive.ubuntu.com/ubuntu/ focal main restricted" >> /etc/apt/sources.list
+            apt update
+            apt install -y ovmf
+            mv /etc/apt/sources.list.back /etc/apt/sources.list
+            apt update
+        fi
+        cp -f /usr/share/OVMF/OVMF_VARS.snakeoil.fd "$WORK_DIR/image/OVMF_VARS.snakeoil.fd"
+        if ! snap list swtpm-mvo; then
+            snap install swtpm-mvo --beta
+        fi
 
-    cp -f "$WORK_DIR/image/ubuntu-core.img" "$IMAGE"
-    systemd_create_and_start_unit "$NESTED_VM" "${QEMU} -m 2048 -nographic \
-        -net nic,model=virtio -net user,hostfwd=tcp::$SSH_PORT-:22 \
-        -drive file=$IMAGE,cache=none,format=raw \
-        -drive file=$WORK_DIR/assertions.disk,cache=none,format=raw \
-        -monitor tcp:127.0.0.1:$MON_PORT,server,nowait -usb \
-        -machine accel=kvm"
+        PARAM_CPU="-smp 2"
+        PARAM_BIOS="-drive file=/usr/share/OVMF/OVMF_CODE.secboot.fd,if=pflash,format=raw,unit=0,readonly=on -drive file=$WORK_DIR/image/OVMF_VARS.snakeoil.fd,if=pflash,format=raw,unit=1"
+        PARAM_IMAGE="-drive file=$IMAGE_FILE,cache=none,format=raw,id=disk1,if=none -device virtio-blk-pci,drive=disk1,bootindex=1"
+        PARAM_MACHINE="-machine q35 -global ICH9-LPC.disable_s3=1"
+        PARAM_TPM="-chardev socket,id=chrtpm,path=/var/snap/swtpm-mvo/current/swtpm-sock -tpmdev emulator,id=tpm0,chardev=chrtpm -device tpm-tis,tpmdev=tpm0"
+    else
+        PARAM_CPU=""
+        PARAM_BIOS=""
+        PARAM_IMAGE="-drive file=$IMAGE_FILE,cache=none,format=raw"
+        PARAM_MACHINE=""
+        PARAM_TPM=""
+    fi
 
+    # Systemdd unit is created, it is important to respect the qemu parameters order
+    systemd_create_and_start_unit "$NESTED_VM" "${QEMU} \
+        ${PARAM_CPU} \
+        ${PARAM_MEM} \
+        ${PARAM_MACHINE} \
+        ${PARAM_DISPLAY} \
+        ${PARAM_NETWORK} \
+        ${PARAM_BIOS} \
+        ${PARAM_TPM} \
+        ${PARAM_IMAGE} \
+        ${PARAM_ASSERTIONS} \
+        ${PARAM_MONITOR} \
+        ${PARAM_EXTRA} "
+
+    # This is a workaround for the issue when the user assertion is not autoimported correctly
     if ! wait_for_ssh; then
         systemctl restart nested-vm
+
+        # This is a workaround for the issue connecting to the swtpm-mvo snap
+        while ! systemctl is-active nested-vm; do
+            systemctl restart nested-vm
```

Comment:
This should use the env var for `nested-vm`, no?

```suggestion
        while ! systemctl is-active "$NESTED_VM"; do
            systemctl restart "$NESTED_VM"
```

Reply:
right, thanks!!!